### PR TITLE
Migrate to Dream

### PR DESCRIPTION
**Pull request by d4hines**
## Depends

- [x] #425

## Problem

We use the `opium` library for defining our http routes. We want to use `dream`. There are two reasons:
- Dream has more modern interface
- Dream's dependencies are more comatible with 5.0

## Solution

This PR does a straightforward swap of opium for dream.

~**Warning!:** There is something wrong the esy setup in this PR. deku wasn't working on my machine except if I installed `libev` with my local package manager, which doesn't seem right. I'd appreciate help debugging why.~

Edit: we patched the development scripts to work by exporting the right LD_LIBRARY_PATH. And for production it won't matter because it's statically linked.

### CODE REVIEW

File: src/bin/deku_node.ml [hunk 1]
```diff
@@ -155,26 +119,28 @@ let node folder =
...
-    |> Lwt_main.run in
-  let forever, _ = Lwt.wait () in
-  Lwt_main.run forever
+  Node.Server.start ~initial:node;
+  Dream.initialize_log ~level:`Warning ();
+  let port = Node.Server.get_port () |> Option.get in
+  Dream.run ~port
+  @@ Dream.router
+       [
+         handle_received_block_and_signature;
```
Comment (gsebil08) [inline src/bin/deku_node.ml:128]: This is one is present two times, is it normal?

File: src/node/flows.ml [hunk 2]
```diff
@@ -9,6 +9,40 @@ let write_state_to_file path protocol =
       Lwt_io.with_file ~mode:Output path (fun oc ->
           let%await () = Lwt_io.write oc protocol_bin in
           Lwt_io.flush oc))
+
+let string_of_error = function
+  | `Added_block_has_lower_block_height -> "Added block has lower block height"
```
Comment (gsebil08) [inline src/node/flows.ml:14]: Should the error printed be more `human friendly`?

Comment (EduardoRFS) [reply src/node/flows.ml:14]: Resolving this, you're right, but there is an open issue for it. https://github.com/marigold-dev/deku/issues/64

File: src/bin/deku_node.ml (hunk 1 above)
Comment (EduardoRFS) [reply src/bin/deku_node.ml]: Nice catch, fixed on a future commit, no, it's not normal but should not be a problem either.

**Review by EduardoRFS — APPROVED** [review]
anime girl thumbs up to a straightforward PR

**Pull request merged**